### PR TITLE
Adaptive progress bar length

### PR DESCRIPTION
**Pull request by innerlee**
not tested yet. wait for a minute before merge

### CODE REVIEW

**Comment by innerlee** [comment]
verified locally.

**Comment by hellock** [comment]
Better to add some unittests for the adaptive behavior.

Another thing to confirm: will the checking in each update affects the fps, especially for a large fps?

**Comment by innerlee** [comment]
image

File: tests/test_progressbar.py [hunk 1]
```diff
@@ -1,7 +1,13 @@
 # Copyright (c) Open-MMLab. All rights reserved.
 import sys
+import os
```
Comment (hellock) [inline tests/test_progressbar.py]: `import os` should be the first statement according to alphabet.

Comment (innerlee) [reply tests/test_progressbar.py]: why does pre-commit not doing this automatically?

**Pull request merged**